### ISSUE TITLE
Specify custom command syntax using vorpal?

### ISSUE THREAD

**Issue by nfischer** [dthree/cash#42]
This is somewhat related to vorpal, but I posted it here in case there's a way to do a parsing work around to leave the vorpal stuff untouched.

Is there a way to specify a custom syntax using vorpal? The syntax I'm interested in is something like `<id>=[value]` (this is what variable reassignment looks like).

If it's possible to write something to recognize that form and preparse it to the form `_assign <id> [value]`, that's probably a good enough work around, since that would let assign be treated like a regular command.

**Comment by dthree** [comment]
I was thinking about just adding `<id>=[value]` as alternate parsing to all commands in Vorpal, as, yeah, its coming up in Cash a lot, and is a very familiar method.

Since I'm about to jump under the arg-parsing hood anyways, I'll just do it in Vorpal.

**Comment by nfischer** [comment]
SGTM. Let me know when it's resolved, and I can probably write up regular variable assignments fairly quickly.

**Comment by dthree** [comment]
:+1:

**Comment by dthree** [comment]
What I did is normalize `key=value` pairs in Vorpal, for now. So it will always return `key='value'`, no matter how the user does it, like:

`foo=bar => foo='bar'`

or

`'foo="bar food cat"' => foo='bar food cat'`

Then its pretty simple for you to arg parse from there.

Couldn't really think of a better way to do it API-wise. Closing this for now, but if you have an API suggestion post it here and I'll re-open.
